command.go

```diff
@@ -317,7 +317,29 @@ func (c *Command) parseFlags(args Args, shellComplete bool) (*flag.FlagSet, erro
 		return set, set.Parse(append([]string{"--"}, args.Tail()...))
 	}
 
-	err = parseIter(set, c, args.Tail(), shellComplete)
+	for pCtx := ctx.parentContext; pCtx != nil; pCtx = pCtx.parentContext {
+		if pCtx.Command != nil {
+			for _, fl := range pCtx.Command.Flags {
+				if pfl, ok := fl.(PersistentFlag); ok && pfl.IsPersistent() {
+					applyPersistentFlag := true
+					set.VisitAll(func(f *flag.Flag) {
+						for _, name := range fl.Names() {
+							if name == f.Name {
+								applyPersistentFlag = false
+							}
+						}
+					})
+					if applyPersistentFlag {
+						if err := fl.Apply(set); err != nil {
+							return nil, err
+						}
+					}
+				}
+			}
+		}
+	}
+
+	err = parseIter(set, c, args.Tail(), ctx.shellComplete)
```

Comment:
I realize this is pre-existing, but I'd prefer this to be combined with the following line as an `if err := ...; err != nil {`

Reply:
Updated